ldap/servers/slapd/filter.c

```diff
@@ -1094,8 +1096,15 @@ slapi_filter_free_bits(Slapi_Filter *f)
 int
 slapi_filter_replace_strfilter(Slapi_Filter *f, char *strfilter)
 {
-    slapi_filter_free_bits(f);
-    return (slapi_filter_replace_ex(f, strfilter));
+    /* slapi_filter_replace_ex may fail and we cannot
```

Comment:
I agree with the fact to check the filter before freeing the bits. Shouldn't we also fix slapi_filter_replace_strfilter ?

Reply:
I do not understand your comment:
The fix is already inside slapi_filter_replace_strfilter and apparently that is the only place where slapi_filter_free_bits is called...

Reply:
Oooppps.... sorry I misread the function name. forget my comment :)